Source/WebKit/_WebKit_SwiftUI/Foundation+Extras.swift

```diff
@@ -35,16 +35,55 @@ func onNextMainRunLoop(do body: @escaping @MainActor () -> Void) {
 }
 
 struct AnyEquatable: Equatable {
-    let value: Any
-    private let equals: (Any) -> Bool
+    private enum Storage {
+        case value(Any, (Any) -> Bool)
+        case never
+    }
+
+    private let storage: Storage
+
+    var value: Any {
+        switch storage {
+        case let .value(value, _): value
+        case .never: ()
+        }
+    }
 
     init<E: Equatable>(_ value: E) {
-        self.value = value
-        self.equals = { ($0 as! E) == value }
+        self.storage = .value(value, { other in
```

Comment:
I think you could perhaps use `AnyHashable` here, if you’re willing to limit the initializer to `some Hashable` instead (technically weird, but it gets you off the hook for a lot of this stuff, and most types you’re likely to use it with should still work then).

Otherwise, you could probably just capture the meta type directly? `case value(Any, Any.Type)` (didn’t test that though).